File: ViewModel_Exam/Java/app/src/main/java/com/example/dgsw/viewmodel_exam/MainActivity.java
package com.example.dgsw.viewmodel_exam;

import android.os.Bundle;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;
import androidx.room.Room;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        EditText mTodoEditText = findViewById(R.id.todo_edit);
        TextView mResultTextView = findViewById(R.id.result_text);

        MainViewModel viewModel = new ViewModelProvider(this).get(MainViewModel.class);


        viewModel.getAll().observe(this, it -> mResultTextView.setText(it.toString()));

        findViewById(R.id.add_button).setOnClickListener(v ->
            viewModel.insert(new Todo(mTodoEditText.getText().toString()))
        );

}
}

File: ViewModel_Exam/Kotlin/app/src/main/java/com/example/dgsw/viewmodel_exam/MainActivity.kt
package com.example.dgsw.viewmodel_exam

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import androidx.room.Room
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import io.reactivex.schedulers.Schedulers.io

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val editText = findViewById<EditText>(R.id.todo_edit)
        val textView = findViewById<TextView>(R.id.result_text)
        val button = findViewById<Button>(R.id.add_button)

        val mainViewModel = ViewModelProvider(this).get(MainViewModel::class.java);


        mainViewModel.getAll().observe (this, { todo ->
                    textView.text = todo.toString()
                }
        ) // LiveData 사용

        button.setOnClickListener {
            mainViewModel.insert(Todo(editText.text.toString()))
        }
    }
}